src/app/public/src/modules/nav/nav.service.ts

```diff
@@ -76,36 +81,40 @@ export class StacheNavService {
 
   private formatRoutes(routes: any[]): StacheNavLink[] {
     let formatted = routes.map(route => {
-      let name = this.getPreferredName(route.path);
+      let pathMetadata = this.getRouteMetadata(route);
 
-      if (name === '') {
-        name = this.getNameFromPath(route.segments[route.segments.length - 1]);
-      }
+      let formattedRoute = Object.assign({},
+        {
+          path: `/${route.path}`,
+          children: this.formatRoutes(route.children),
+          name: this.getNameFromPath(route.segments[route.segments.length - 1])
+        },
+        pathMetadata);
 
-      return {
-        name: name,
-        path: `/${route.path}`,
-        children: this.formatRoutes(route.children)
-      } as any;
+      return formattedRoute as any;
     });
 
-    return formatted as StacheNavLink[];
+    return this.sortRoutes(formatted) as StacheNavLink[];
   }
 
-  private getPreferredName(path: string): string {
-    const routeMetadata = this.routeMetadataService.routes;
+  private getRouteMetadata(route: any): StacheNavLinkMetadata {
+    const allMetadata = this.routeMetadataService.routes;
+    let pathMetadata: StacheNavLinkMetadata = {};
 
-    if (routeMetadata) {
-      let foundRoute = routeMetadata.filter((route: any) => {
-        return route.path === path;
+    if (allMetadata) {
+      let foundRoute = allMetadata.filter((metaRoute: any) => {
+        return metaRoute.path === route.path;
       })[0];
 
       if (foundRoute) {
-        return foundRoute.name;
+        if (foundRoute.order !== undefined) {
+          foundRoute.order = parseInt(foundRoute.order, 10);
+        }
+        pathMetadata = Object.assign({}, pathMetadata, foundRoute);
```

Comment:
I might be missing something; is this `Object.assign` unnecessary? `pathMetadata` will always be `{}` at this moment.